src/Components/Modal/modal.js

```diff
@@ -1,24 +1,100 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./modal.css";
 import ModalItems from "./Modal-Items/ModalItems";
 
 const Modal = () => {
-	return (
-		<div className="modal-class">
-			<p>Available Routes</p>
-			<span>
-				<label>Direction</label>
-				<label>Duration</label>
-				<label>Vehicle</label>
-				<label>Cost</label>
-			</span>
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-		</div>
-	);
+	const [routes, setRoutes] = useState([]);
+
+	const fetchRoutes = () => {
+		let requestBody = {
+			query: `
+         query{
+            routes{
+              id
+              direction
+              vehicle
+              duration
+              cost
+            }
+       }
+      `,
+		};
+
+		fetch("https://ride-booking-app.herokuapp.com/graphql", {
+			method: "post",
+			body: JSON.stringify(requestBody),
+			headers: {
+				"Content-Type": "application/json",
+			},
+		})
+			.then((res) => {
+				if (res.status !== 200 && res.status !== 201) {
+					throw new Error("Authentication Failed");
```

Comment:
how do you intend to handle this error from the perspective of the user? How will they know auth has failed?